esmvaltool/diag_scripts/ipcc_ar6/corr_pattern_collect.ncl

```diff
@@ -0,0 +1,481 @@
+                  ymin = yy(0)
+                end if
+                if (yy(0) .gt. ymax) then
+                  ymax = yy(0)
+                end if
+            end if
+          end if
+        end do
+        if (diag_script_info@plot_median) then
+          plot@$unique_string("dum_ci")$ = gsn_add_polymarker( \
+            wks, plot, x_val_proj(iproj, ivar), median(iproj, ivar), res_circw)
+        end if
+        if (isatt(diag_script_info, "model_spread")) then
+          if (diag_script_info@model_spread) then
+            yyyy = (/ymin, ymin, ymax, ymax/)
+            xxxx = (/xx(0), xx(1), xx(1), xx(0)/)
+            res_shading = res_lines
+            res_shading@gsFillColor = fcolors(iproj, :)
+            res_shading@gsFillOpacityF = 0.2
+            plot@$unique_string("dum")$ = gsn_add_polygon( \
+              wks, plot, xxxx, yyyy, res_shading)
+          end if
+        end if
+      end if
+      delete(proj_mods)
+    end do
+    if (alt_obs(ivar).ne."none") then
+      ; Plot obs as circles
+      plot@$unique_string("dum_circ")$ = gsn_add_polymarker( \
+        wks, plot, x_val(ivar), data_all(obs_ind(ivar), ivar), res_circ)
+    end if
+  end do
+
+  y_min_label = res@trYMinF + 0.1*yspan
+  lb_stride = yspan/res@vpHeightF * font_height * 2.
+  plabel = projects
+  ; Draw Legend
+  do iproj = 0, dimsizes(projects)-1
+    res_text@txFontColor = fcolors(iproj, :)
+    ; CMIP5 label has to be reduced to CMIP5 sometimes
+    if str_match_ind_ic(plabel, "CMIP5").eq.iproj then
+      plabel(iproj) = "CMIP5"
+    end if
+    plot@$unique_string("dum_l")$ = gsn_add_text(wks, plot, plabel(iproj),\
+                                                 x_val(0)-0.5, y_min_label + \
+                                                 lb_stride*(iproj+1.2), \
+                                                 res_text)
+  end do
+  res_text@txFontColor = res_circ@gsMarkerColor
+  plot@$unique_string("dum_l")$ = gsn_add_text(wks, plot, \
+                                               "additional Observations", \
+                                               x_val(0)-0.5, y_min_label, \
+                                               res_text)
+
+  draw(plot)
+  frame(wks)
+  ; Write output
+  system("mkdir -p " + config_user_info@work_dir)
+  workpath = config_user_info@work_dir + "pattern_cor.nc"
+  ncdf_outfile = ncdf_write(data_all, workpath)
+
+  ; collect meta-data and call ESMValMD function
+  caption = "Centered pattern correlations between models and observations" \
+            + " for the annual mean climatologies " \
+            + "(similar to IPCC ch. 9 fig. 9.6)."
+  statistics = (/"corr", "clim"/)
+  domains = (/"global"/)
+  plottype = "other"
+  authors = (/"gier_bettina", "bock_lisa"/)
+  references = (/"flato13ipcc"/)
+  log_provenance(ncdf_outfile, outfile, caption, statistics, domains, \
+                 plottype, authors, references, prov_files)
+  leave_msg(DIAG_SCRIPT, "")
+end
```

Comment:
Again as in the corr_pattern.ncl, the changes seem mostly cosmetical, adding some features and thus shouldn't break the recipes with the old diagnostics, would just require to add the `plot_median` in the cfg to get a similar looking plot to before.